src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer.Common/Serialization/JsonConverterCollectionExtensions.cs

```diff
@@ -31,6 +35,67 @@ public static void RegisterRazorConverters(this JsonConverterCollection collecti
             {
                 collection.Add(RazorConverters[i]);
             }
+
+            DelegatingSupportsConverter.AddDelegatingSupportsConverter(collection);
+        }
+    }
+
+    internal class DelegatingSupportsConverter : JsonConverter
+    {
+        private JsonConverter Converter { get;}
+
+        public static void AddDelegatingSupportsConverter(JsonConverterCollection collection)
+        {
+            Type supportsConverterType = typeof(PublishDiagnosticsCapability).Assembly.GetType("OmniSharp.Extensions.LanguageServer.Protocol.Serialization.Converters.SupportsConverter");
```

Comment:
hahah, oh man this be ugly

Reply:
The whole `DelegatingSupportsConverter` class is brought mostly straight from [Todd's PR](https://devdiv.visualstudio.com/DevDiv/_git/WebTools/pullrequest/238775?isPreview=false&path=%2Fsrc%2FLanguages%2Fhtml%2FLanguageServer.InProc%2FHtml%2FHtmlLanguageServer.cs&_a=files). Still, if you have a suggestion for cleaning this up I'm all ears.

Reply:
It's fine if we contain the ugliness. Was just mentioning it is all 😄

Reply:
This ugliness took quite a while to figure out, unfortunately :(